tests/test_stateupdate_authority.c

```
@@ -0,0 +1,138 @@
+    if (!bc_buf_write_cf16(&fb, speed)) return -1;
+
+    return bc_build_state_update(out, out_size,
+                                 object_id, 42.0f,
+                                 (u8)(BC_DIRTY_POSITION_ABS |
+                                      BC_DIRTY_ORIENT_FWD |
+                                      BC_DIRTY_ORIENT_UP |
+                                      BC_DIRTY_SPEED),
+                                 fields, (int)fb.pos);
+}
+
+TEST(stateupdate_downstream_is_server_shaped)
```

Clang (14.0.6)

[warning] function 'test_stateupdate_downstream_is_server_shaped' has cognitive complexity of 66 (threshold 25)

(readability-function-cognitive-complexity)